### PR TITLE
lib/osutil: Fix TraversesSymlink with symlinked fs root on windows (fixes #4875)

### LINKED ISSUE #4875: Zeroed local/global "state" (local files untouched) and remote file deletion with root Windows symlink
When specifying a sync directory as a symlink to another drive and using the filesystem watcher I receive the following error on Windows:

Failed to start filesystem watcher for folder "Documents" (mqanr-z452q): GetFileAttributesEx /S:\Documents: The filename, directory name, or volume label syntax is incorrect.

The path provided in Syncthing's config is "~/Documents", I suspect the / is not being stripped given GetFileAttributesEx **/S:\...** I verified the ~ is not related to the issue by also specifying the absolute path and verifying the error persists.

Also triggered by this is that after a seemingly successful scan of the directory and it's contents (e.g. the number of files and folders look correct) the files and folders drop to 0, and remote devices empty their drives resulting is data loss on remote devices (note the local sync folders are unaffected).

### Version Information

Syncthing Version: v0.14.46
OS Version: Windows 10

### PR DESCRIPTION
**Pull request by imsodin**
It's windows specific - I'll explain if the build server confirms it, nothing to do for anyone here (yet) :)

### CODE REVIEW

**Comment by imsodin** [comment]
In #4765 (v0.14.46) I started using `osutil.TraversesSymlink` (through `osutil.IsDeleted`) to detect deleted files. This is fine except that `osutil.TraversesSymlink` wrongly also checks the folder root. It uses `IsDir`, which on linux is true for a symlink to a dir, but false on windows. So if you change the folder root to be a symlink on windows, the entire folder gets wiped in the db. So this was nicely hidden by working on linux as expected and there being hardly any windows&symlink tests - proving once again that the combination is evil.

This is in principle no rc material, but given that it has potential for data loss, I'd propose to still add it to v0.14.47 @AudriusButkevicius @calmh

**Comment by AudriusButkevicius** [comment]
There not being tests does not prove the combination is evil, it proves we are shit developers.

File: lib/osutil/traversessymlink.go [hunk 1]
```diff
@@ -32,29 +32,21 @@ func (e NotADirectoryError) Error() string {
...
 	}
-	if !info.IsDir() {
-		return &NotADirectoryError{
-			path: base,
-		}
-	}
 
 	if name == "." {
-		// The result of calling TraversesSymlink("some/where", filepath.Dir("foo"))
 		return nil
```
Comment (AudriusButkevicius) [inline lib/osutil/traversessymlink.go]: Leave this comment?

**Comment by calmh** [comment]
LGTM. So with this in mind, maybe we can make an attempt to fix the following symlink related tests that are skipped on Windows (maybe not *all* of them symlink related, but clearly many):

screen shot 2018-04-17 at 19 28 03

**Comment by imsodin** [comment]
Hooks seem to be broken and manually starting builds runs on the old commits - so fuck tests and merge? It was just a comment.

**Comment by AudriusButkevicius** [comment]
Do it

**Pull request merged**